### DISCUSSION TITLE
What is the best way to update a zarr archive

### DISCUSSION THREAD

**Discussion by Boorhin** [pydata/xarray#6700] [Q&A] [answered]
I am producing some zarr archive of xarray datasets and I would like to compute new variables and add them to the same archive without modifying what is already existing. What is the best practice?
```
ds=xr.open_zarr('archive.zarr', mode='a')
ds['new_var']=ds['var1']-ds['var2']
```
=> will the `'new_var'` saved automatically or do I have to force the writing of it with `to_zarr`?
`ds.to_zarr('archive.zarr')`
Thanks

**Comment by shoyer** [comment, marked as answer]
Xarray datasets are never directly linked to data on disk. To modify an existing archive, you need to explicitly call `to_zarr()` or `to_netcdf()`.
